src/main/java/uni/miskolc/ips/ilona/measurement/controller/MeasurementController.java

```diff
@@ -19,105 +20,102 @@
-
-  /** Reads data from context.xml automatically. */
   private final MeasurementService measurementManagerService;
 
-  /**
-   * Lists the available measurements based on the zoneID which is not necessarily required.
-   *
-   * @param zoneId The zoneID of the list is based on
-   * @return Returns the list of results
-   */
   @GetMapping(value = {"", "/"})
-  public List<MeasurementDTO> listMeasurements(
-      @RequestParam(value = "zoneId", required = false) final UUID zoneId)
-      throws DatatypeConfigurationException, DatabaseUnavailableException {
-    List<MeasurementDTO> result = new ArrayList<>();
-    Collection<Measurement> measurements = measurementManagerService.readMeasurements();
-    for (Measurement measurement : measurements) {
-      if (zoneId != null && !measurement.getPosition().getZone().getId().equals(zoneId)) {
-        continue;
+  public List<MeasurementDTO> listMeasurements(@RequestParam(value = "zoneId", required = false) UUID zoneId) {
+    try {
+      List<MeasurementDTO> result = new ArrayList<>();
+      Collection<Measurement> measurements = measurementManagerService.readMeasurements();
+      for (Measurement measurement : measurements) {
+        if (zoneId != null && !measurement.getPosition().getZone().getId().equals(zoneId)) {
+          continue;
+        }
+        result.add(assembleMeasurementDTO(measurement));
       }
-      result.add(assembleMeasurementDTO(measurement));
+      return result;
+    } catch (DatatypeConfigurationException e) {
+      log.info(e.getMessage());
+      throw new ResponseStatusException(HttpStatus.UNSUPPORTED_MEDIA_TYPE, e.getMessage());
+    } catch (DatabaseUnavailableException e) {
+      log.info(e.getMessage());
+      throw new ResponseStatusException(HttpStatus.INSUFFICIENT_STORAGE, e.getMessage());
     }
-    return result;
   }
 
-  /**
-   * Calls the measurement manager service to create a measurement. It accepts only post requests.
-   *
-   * @param measurementRegistrationRequest measurement data
-   */
   @PostMapping(value = "/")
-  public void recordMeasurement(
-      @RequestBody final MeasurementRegistrationRequest measurementRegistrationRequest)
-      throws InconsistentMeasurementException, DatabaseUnavailableException {
-    Measurement measurement = new Measurement();
-    measurement.setId(UUID.randomUUID());
-    measurement.setTimestamp(new Date());
-    measurement.setPosition(dispersePositionDTO(measurementRegistrationRequest.getPosition()));
-    if (measurementRegistrationRequest.getWifiRSSI() != null) {
-      Map<String, Double> wifiRssis = new HashMap<>();
-      for (MeasurementRegistrationRequest.WifiRSSI.Ap ap :
-          measurementRegistrationRequest.getWifiRSSI().getAp()) {
-        wifiRssis.put(ap.getSsid(), ap.getValue());
+  public void recordMeasurement(@RequestBody MeasurementRegistrationRequest measurementRegistrationRequest) {
+    try {
+      Measurement measurement = new Measurement();
```

Comment:
move the initialization of Measurement into a private method

private Measurement convertMeasurementRecordRequest2Measurement(MeasurementRegistrationRequest req);

It will simplify this controller

Reply:
Not necessaray see #32